google-cloud-pubsub/test/google/cloud/pubsub/async_publisher_test.rb

```diff
@@ -344,6 +344,8 @@
 
     callbacks.must_equal 2
 
+    sleep 0.1
```

Comment:
Right. The thing is, we have the blocking check in the method `wait!`. I suspect that the problem here is actually with the Mock object. But I don't know for sure. I cannot reproduce locally, so I'm going to use this PR to try to reproduce on CI.